### ISSUE TITLE
HPA Feature: Ability to select workloads via label

### ISSUE THREAD

**Issue by bentastic27** [kubernetes/kubernetes#94615]
**What would you like to be added**:

It would be useful to be able to select workloads via label instead of just the [object reference](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#crossversionobjectreference-v2beta2-autoscaling) in the `scaleTargetRef` field in the spec. The desire is to essentially have a single HPA definition that covers multiple workloads.

**Why is this needed**:

Basically looking for a way to have a single HPA definition to apply to multiple workloads. Having 500 deployments would currently require 500 HPAs. Being able to apply by label would greatly reduce this and make it much more manageable.

**Comment by umialpha** [comment]
Hi I also need this feature. Do you have some work around?
